### ISSUE TITLE
Swagger-UI template does not includes oauth2-redirect.html

### ISSUE THREAD

**Issue by Colin-b** [luolingchun/flask-openapi3#159]
Hi,

OAuth2 flows requiring a redirect (such as authorization code) are not working because Swagger-UI will request the token to be received on oauth2-redirect.html, and this file does not exists:
![issue_redirect](https://github.com/luolingchun/flask-openapi3/assets/7591933/a6fa5719-c3cf-40a8-8bef-e508c7a4c9c5)

Looking at the new plugins repository that you introduced with the future version 4, it looks like a new endpoint will be added (not sure if it will serves the .html though as it seems to be on /oauth2-redirect and not /oauth2-redirect.html).

Can you confirm that this is a known issue with v3 and v4 will fix it? Or should we expect a fix in v3?

Thanks again

Environment:

- Python version: 3.11.8
- Operating system: Linux
- Flask version: 2.2.5
- flask-openapi3 version: 3.1.3

**Comment by luolingchun** [comment]
This feature is indeed implemented in the v4 release, using the route `/openapi/swagger/oauth2-redirect`.

The v4 ([v4.0.0rc2](https://pypi.org/project/flask-openapi3/4.0.0rc2/)) version is initially available, but there are some breaking updates, and the documentation is not ready. Here's how to use this feature:

```python
from flask_openapi3 import Info
from flask_openapi3 import OpenAPI

info = Info(title='oauth API', version='1.0.0')

# https://spec.openapis.org/oas/v3.1.0#implicit-oauth2-sample
oauth2 = {
    "type": "oauth2",
    "flows": {
        "implicit": {
            "authorizationUrl": "/openapi/swagger/oauth2-redirect",
            "scopes": {
                "write:pets": "modify pets in your account",
                "read:pets": "read your pets"
            }
        }
    }
}
security_schemes = {"oauth2": oauth2}

app = OpenAPI(__name__, info=info, security_schemes=security_schemes)

# https://github.com/swagger-api/swagger-ui/blob/0ce792c9d0965a8b6b5d75f5e1341ff6936a4cb0/docs/usage/oauth2.md
oauth_config = {
    "clientId": "xxx",
    "clientSecret": "xxx"
}
app.config["OAUTH_CONFIG"] = oauth_config

# https://spec.openapis.org/oas/v3.1.0#oauth2-security-requirement
security = [
    {"oauth2": ["write:pets", "read:pets"]}
]


@app.get("/", security=security)
def oauth():
    return "oauth"


if __name__ == '__main__':
    app.run(debug=True)
```

**Comment by Colin-b** [comment]
Thanks for the quick reply,

I think we are mixing concerns here.

The authorization URL is a parameter related to the OAuth2 flow, this is not something you can really tweak as this is supposed to be related to your identity provider.

Looking at Swagger-UI documentation, there is a [oauth2RedirectUrl](https://github.com/swagger-api/swagger-ui/blob/HEAD/docs/usage/configuration.md#network) parameter that can be [provided to the SwaggerUI constructor](https://github.com/swagger-api/swagger-ui/blob/HEAD/docs/usage/configuration.md#how-to-configure), and I assume the same applies to the [SwaggerUIBundle](https://github.com/luolingchun/flask-openapi3-plugins/blob/4a6a0e10067a18396e5461019b5a1f3c6ad182fa/flask-openapi3-swagger/flask_openapi3_swagger/templates/__init__.py#L38) parameter that you are using in the plugins. 
Meaning this parameter can be provided via the [SWAGGER_CONFIG](https://github.com/luolingchun/flask-openapi3-plugins/blob/4a6a0e10067a18396e5461019b5a1f3c6ad182fa/flask-openapi3-swagger/flask_openapi3_swagger/plugins.py#L34) flask configuration key. 
It looks like no default value for this setting is set by `flask-openapi3`, so it could in theory be provided. My only concern is that this URL will have to be a full one, and not a path that would be appended afterwards. You can see that [this is what is done for the default value in Swagger-UI](https://github.com/swagger-api/swagger-ui/blob/0ce792c9d0965a8b6b5d75f5e1341ff6936a4cb0/src/core/config/sources/runtime.js#L12).

To sum this up, I think it should be the responsability of either flask-openapi3, either the flask-openapi3 swagger-ui plugin to set this value properly.

In the meantime I will see if I can try this out on my end by providing a `https://my_server/openapi/swagger/oauth2-redirect` type of URL.

**Comment by luolingchun** [comment]
> An other solution (and maybe a cleaner one) would be to serve the redirect page in the expected Swagger-UI default route. I created a PR for this.


The PR above has been merged, now please upgrade [flask-openapi3-swagger](https://pypi.org/project/flask-openapi3-swagger/) to the latest version

**Comment by Colin-b** [comment]
I confirm the redirection works now !

Thanks again for the reactivity
